relentless/optimize/method.py

```diff
@@ -177,41 +193,67 @@ def __init__(self, abs_tol, max_iter):
         self.abs_tol = abs_tol
         self.max_iter = max_iter
 
-    def find(self, objective, start, end):
+    def find(self, objective, start, end, grad_scale=1.0):
         """Apply the line search algorithm to take the optimal step.
 
         Note that the objective function is kept at its initial state, and the
         function evaluted after taking the optimal step is returned separately.
 
         Parameters
         ----------
-        objective : :class:`~relentless.optimize.ObjectiveFunction`
+        objective : :class:`~relentless.optimize.objective.ObjectiveFunction`
             The objective function for which the line search is applied.
-        start : :class:`~relentless.optimize.ObjectiveFunctionResult`
+        start : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
             The objective function evaluated at the start of the search interval.
-        end : :class:`~relentless.optimize.ObjectiveFunctionResult`
+        end : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
             The objective function evaluated at the end of the search interval.
+        grad_scale : float or dict or :class:`~relentless._collections.KeyedArray`
+            The gradient scaling parameter or parameters (:math:`\mathbf{b}`)
+            keyed on the :class:`~relentless.optimize.objective.ObjectiveFunction`
+            design variables (defaults to ``1.0``).
 
         Raises
         ------
+        ValueError
+            If the gradient scaling parameters are not positive.
```

Comment:
@zeesherman Do they need to be positive? It seems like a negative value would be allowable, even if it's silly.